config/v1/types_infrastructure.go

```
@@ -600,9 +600,300 @@ type OvirtPlatformStatus struct {
+	// case it is expected that region would be a cluster.
+	// HostGroups within vCenter must be preconfigured and
+	// assigned in the topology.
+	HostGroupFailureDomain VSpherePlatformFailureDomainType = "HostGroup"
+
+	// ComputeClusterFailureDomain failure domain can either be a zone or region.
+	// The vCenter cluster is required to preconfigured and
+	// assigned in the topology.
+	ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
+
+	// DatacenterFailureDomain failure domain can be only be a region. The vcenter
+	// datacenter is required to be preconfigred and assigned
+	// in the topology. If used the zone would be of type ComputeCluster.
+	DatacenterFailureDomain VSpherePlatformFailureDomainType = "Datacenter"
+)
+
+// VSpherePlatformFailureDomainSpec holds the region and zone failure domain and
+// the vCenter topology of that failure domain.
+type VSpherePlatformFailureDomainSpec struct {
+	// name defines the name of the VSpherePlatformFailureDomainSpec
+	// This name is arbitrary but will be used
+	// in VSpherePlatformDeploymentZone for association.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"name"`
+
+	// region defines a VSpherePlatformFailureDomain which
+	// includes the name of the vCenter tag, the failure domain type
+	// and the name of the vCenter tag category.
+	// +kubebuilder:validation:Required
+	Region VSpherePlatformFailureDomain `json:"region"`
+
+	// zone defines a VSpherePlatformFailureDomain which
+	// includes the name of the vCenter tag, the failure domain type
+	// and the name of the vCenter tag category.
+	// +kubebuilder:validation:Required
+	Zone VSpherePlatformFailureDomain `json:"zone"`
+
+	// Topology describes a given failure domain using vSphere constructs
+	// +kubebuilder:validation:Required
+	Topology VSpherePlatformTopology `json:"topology"`
```

I'm getting the impression that the topology should be a discrminated union

```
topology:
  type: Datacenter
  datacenter:
    name: ...
```

Why is the type in the region/zone failure domain rather than the topology?

---

With the removal of HostGroup as a zone not sure this still applies.

Topology defines the vCenter-based objects that a region-zone pair includes.